eth/beacon/helpers.py

```diff
@@ -408,38 +408,77 @@ def get_domain(fork_data: 'ForkData',
     ) * 4294967296 + domain_type
 
 
-def verify_slashable_vote_data(state: BeaconState,
-                               vote_data: 'SlashableVoteData',
-                               max_casper_votes: int) -> bool:
+@to_tuple
+def get_pubkey_for_indices(validators: Sequence[ValidatorRecord],
+                           indices: Sequence[int]) -> Iterable[int]:
+    for index in indices:
+        yield validators[index].pubkey
+
+
+@to_tuple
+def generate_aggregate_pubkeys(state: BeaconState, vote_data: 'SlashableVoteData') -> Iterable[int]:
+    """
+    Computes the aggregate pubkey we expect based on
+    the proof-of-custody indices found in the ``vote_data``.
+    """
     proof_of_custody_0_indices = vote_data.aggregate_signature_poc_0_indices
     proof_of_custody_1_indices = vote_data.aggregate_signature_poc_1_indices
-    vote_count = len(proof_of_custody_0_indices) + len(proof_of_custody_1_indices)
-    if vote_count > max_casper_votes:
-        return False
+    all_indices = (proof_of_custody_0_indices, proof_of_custody_1_indices)
+    get_pubkeys = functools.partial(get_pubkey_for_indices, state.validators)
+    return map(bls.aggregate_pubkeys,
+               map(get_pubkeys, all_indices))
 
-    pubkeys = [
-        bls.aggregate_pubkeys([state.validators[i].pubkey for i in proof_of_custody_0_indices]),
-        bls.aggregate_pubkeys([state.validators[i].pubkey for i in proof_of_custody_1_indices])
-    ]
 
+def build_vote_data_messages(vote_data: 'SlashableVoteData') -> Tuple[bytes, bytes]:
+    """
+    Builds the messages that validators are expected to sign for a ``CasperSlashing`` operation.
+    """
     # TODO: change to hash_tree_root(vote_data) when we have SSZ tree hashing
     vote_data_root = vote_data.root
-    messages = [
-    ]
+    return (
         vote_data_root + (0).to_bytes(1, 'big'),
         vote_data_root + (1).to_bytes(1, 'big'),
+    )
+
+
+def verify_vote_count(vote_data: 'SlashableVoteData', max_casper_votes: int) -> bool:
+    """
+    Ensures we have no more than ``max_casper_votes`` in the ``vote_data``.
+    """
+    return vote_data.vote_count <= max_casper_votes
+
+
+def verify_signature(state: BeaconState, vote_data: 'SlashableVoteData') -> bool:
```

Comment:
i feel like this function name is fairly generic given what it is doing -- either we should specialize the function name or generalize the arguments (which really just gets us to `bls.verify_multiple` so not ideal); this function was named `verify_signature_for_slashable_vote_data` but the linter did not like the line length on 479.

anyone have any preference or ideas on this?

Reply:
`verify_slashable_vote_data_signature`?
For line 479, you can do:
```python
    return (
        verify_vote_count(vote_data, max_casper_votes) and
        verify_slashable_vote_data_signature(state, vote_data)
    )
```